Cargo.toml

```diff
@@ -16,10 +16,15 @@ license = "Apache-2.0"
 [workspace.dependencies]
 indexmap.version = "2.2.6"
 hashbrown.version = "0.14.0"
-# This doesn't set `extension-module` as a shared feature because we need to be able to disable it
-# during Rust-only testing (see # https://github.com/PyO3/pyo3/issues/340).
+# Most of the crates don't need `extension-module`, since only `qiskit` builds an actual C extension.
```

Comment:
I'm alright to remove it if you think it's unhelpful, but I think what we've got here is a slightly more unusual structure where the separate 4 crates still use all the PyO3 `pyclass` and `pyfunction` stuff, but deliberately _without_ linking `libpython`.  To me, that feels like an unusual use of PyO3 that I thought deserved a comment as to why it's not an inherited feature along with `abi3`.

Reply:
That's fair. Can you perhaps add a link to the docs for `extension-module` (https://pyo3.rs/main/features#extension-module) or briefly describe what it is, so it's clearer why it's interesting that it's missing?

(Even as a PyO3 user, I didn't really know what it was since I don't frequently reconfigure PyO3 crate features)

Reply:
Fixed in ed2ff2c776.